clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1354,8 +1373,21 @@ public void handle(OffsetCommitResponse commitResponse, RequestFuture<Void> futu
             Set<String> unauthorizedTopics = new HashSet<>();
 
             for (OffsetCommitResponseData.OffsetCommitResponseTopic topic : commitResponse.data().topics()) {
+                String topicName = topic.name();
+
+                if (this.response.requestHeader().apiVersion() >= 9) {
+                    topicName = topicIdAndNames.topicNameOrNull(topic.topicId());
+
+                    if (topicName == null) {
+                        // Could only happen if the broker replied with an ID which was not in the request and
+                        // unknown by this client. This would be a bug.
+                        log.warn("Ignoring invalid topic ID found in OffsetCommit response: " + topic.topicId());
+                        continue;
+                    }
```

Comment:
As a second thought, I wonder if we should complete the future with an exception here. Being defensive would help us to catch bugs early one. What do you think?